src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1898,6 +1908,19 @@ private function registerHttpClientConfiguration(array $config, ContainerBuilder
                     ->setArguments([new Reference('http_client'), [$scope => $scopeConfig], $scope]);
             }
 
+            if ($debug) {
```

Comment:
It could be interesting to add a tag to the service a this point, and to do the wiring logic in a compiler pass for all services having this tag. It will allow bundles to easily plug their own clients in the profiler.

For instance, we have this use case in API Platform.

Reply:
I'm almost done except your comment.

To be sure to get your point I looked the Messenger stuff. So we are ok to add a tag to all http client here https://github.com/symfony/symfony/blob/4.4/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L1946-L1953
Like `http_client.client` ?

Then adding a HttpClientPass to make the call to the dataCollector in all service tagged with `http_client.client`. (if data_collector is registered of course).

Right ?